### PR TITLE
remove the rootID in the tree

### PR DESCRIPTION
**Pull request by ryanfkeepers**
Instead of comparing IDs, we can check the drive item's GetRoot() property, which is only non-nil for the root folder.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :bug: Bugfix

#### Issue(s)

* #4689

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/m365/collection/drive/delta_tree.go [hunk 1]
```diff
@@ -336,32 +335,33 @@ func (face *folderyMcFolderFace) addFile(
...
 
 	// in case of file movement, clean up any references
 	// to the file in the old parent
 	oldParentID, ok := face.fileIDToParentID[id]
 	if ok && oldParentID != parentID {
-		if nodey, ok := face.folderIDToNode[oldParentID]; ok {
+		if nodey := face.getNode(oldParentID); nodey != nil {
 			delete(nodey.files, id)
 		}
-
```
Comment (pandeyabs) [inline src/internal/m365/collection/drive/delta_tree.go:356]: Is this code removal intentional? Also this change doesn't look related to `rootID` removal, please correct me if wrong.

Going forward, can we please move unrelated changes to a different PR? :)

**Review by pandeyabs — APPROVED** [review]
LGTM, left a comment to sanity check on an unrelated code removal.

File: src/internal/m365/collection/drive/delta_tree.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/m365/collection/drive/delta_tree.go:356]: Oh, it's all related at this point.  And, yes, this is intentional.

Comment (pandeyabs) [reply src/internal/m365/collection/drive/delta_tree.go:356]: Thanks for clarifying
